myFitnessTrainer/src/components/GenerateNewPlanAlgoButton.js

```diff
@@ -0,0 +1,130 @@
+import { StyleSheet, Text, TouchableOpacity } from "react-native";
+import React from "react";
+import { useNavigation } from "@react-navigation/native";
+import { collection, getDocs, query, where, and } from "firebase/firestore";
+import { db } from "../../firebaseConfig";
+import { auth } from "../../firebaseConfig";
+import { getDocumentId, postDocument } from "../../databaseFunctions";
+
+/*
+    Pass in the following props as the below types:
+    - goal: Array<String>
+    - level: String
+    - duration: Int
+    - modifications: Array<String>
+*/
+const GenerateNewPlanAlgoButton = ({ goal, level, duration }) => {
+    // Test props for debugging
+    goal = goal ?? "strength";
+    level = level ?? "beginner";
+    duration = duration ?? 5;
+    // modifications = modifications ?? ["legs"];
+
+    const navigation = useNavigation();
+
+    /*
+    performs query to db for exercises collection based on goal and level survey result and returns an array of exercises
+    */
+    const getExercises = async (goal, level) => {
+        const q = query(
+            collection(db, "exercises"),
+            filterExerciseQuery(goal, level)
+        );
+        const querySnapshot = await getDocs(q);
+        const exercises = [];
+        querySnapshot.forEach((doc) => {
+            exercises.push(doc.data());
+        });
+        return exercises;
+    };
+
+    const createPlan = (exercises, duration) => {
+        const workoutPlan = [];
+
+        for (let i = 0; i < exercises.length && i < duration; i++) {
+            let randomIdx = Math.floor(Math.random() * exercises.length);
+            workoutPlan.push(exercises[randomIdx]);
+        }
+        return workoutPlan;
```

Comment:
Right now, in the workoutPlan created with this also, we will have a list of 5 different exercises. I think this is almost complete but we should add one more layer to it. 

1) In the create workout survey (I plan on making it this week), we should ask how many exercises they would like to complete _per day_, in addition for the duration of the overall plan. With maybe 4 for the default?

2) This would mean another parameter for this button, we could call it `exercises_per_day` 

3) With this new parameter, we would then have a list of lists with the exercises for the `workoutPlan`.

for example

```
[
[{day0exercise0} ,{day0exercise1}, {day0exercise2}, {day0exercise3}],
[{day1exercise0} ,{day1exercise1}, {day1exercise2}, {day1exercise3}],
[...]
]
```

Reply:
The data structure has been changed. The Firestore api method addDoc() doesn't support nested arrays so I had to make a slight adjustment. Instead of nested arrays, we have a javascript object with key value pairs as: 
{
  day0: [{}, {},..], 
  day1: [{}, {}, ...], .
  ...
}

Reply:
good catch! That works!